Refactor availability controller

hackmastera opened the pull request:
This is a refactor pulling business logic out of the availability controller, moving it with its tests to a `VoyagerLookup` class.

This is work toward making space for the alma side of this functionality, as well as being a step toward #266.

Available for review now or I can work on top of this to get 266 implemented; it just seemed like it might be nice to get it in sooner rather than later.

carolyncole reviewed (approved):
@hackmastera It was hard to determine if your changes had broken anything since the tests were also changed so drastically.  I went ahead and reverted your changes to the controller test and was happy to say they all still passed.  Looks good to me.  I will not merge since @christinach wants to review.

christinach commented on spec/models/voyager_lookup_spec.rb:
```diff
@@ -0,0 +1,324 @@
+        expect(details[:status]).to include('Order Received')
+      end
+    end
+
+    context 'when given a record with an elf (electronic) location' do
+      it 'returns a status of online' do
```
This should say `elf1` not `elf`

hackmastera commented:
> @hackmastera It was hard to determine if your changes had broken anything since the tests were also changed so drastically. I went ahead and reverted your changes to the controller test and was happy to say they all still passed. Looks good to me. I will not merge since @christinach wants to review.

Awesome; thank you for double-checking.

christinach reviewed (changes requested):
It looks great. I added some comments.

christinach commented on spec/models/voyager_lookup_spec.rb:
```diff
@@ -0,0 +1,324 @@
+    end
+  end
+
+  describe '#multiple_bib_availability' do
+    it 'provides partial availability' do
+      bib_id = '929437'
+      allow(VoyagerHelpers::Liberator).to receive(:get_availability).and_return(bib_id => {})
```
This could also be added in a different PR.

hackmastera commented:
@christinach thanks for the review; I believe I've addressed the comments. ready for another look.
